src/rgw/rgw_rest.cc

```diff
@@ -657,6 +657,10 @@ void end_header(struct req_state* s, RGWOp* op, const char *content_type,
 
   dump_trans_id(s);
 
+  if(s->prot_flags & RGW_REST_SWIFT) {
+    dump_header(s, "X-Openstack-Request-Id", s->trans_id);
+  }
+
```

Comment:
@cbodley i did think put the code inside dump_trans_id(), but not sure in the future OpenStack will extend or refine the parameter or not.